build.rs

```diff
@@ -434,3 +448,41 @@ mod c {
         cfg.compile("libcompiler-rt.a");
     }
 }
+
+#[cfg(feature = "c-system")]
+mod c_system {
+    use std::env;
+    use std::process::Command;
+    use std::str;
+    use sources;
+
+    /// Link against system clang runtime libraries
+    pub fn compile(llvm_target: &[&str]) {
+        let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+
+        let llvm_config = env::var("LLVM_CONFIG").expect("LLVM_CONFIG not set");
+        let (subpath, libname) = match target_os.as_str() {
+            "linux" => ("linux", format!("clang_rt.builtins-{}", llvm_target[0])),
+            "macos" => ("darwin", "clang_rt.builtins_osx_dynamic".to_string()),
+            _ => panic!("unsupported target os: {}", target_os),
+        };
+        let cmd = format!("ls -1d $({} --libdir)/clang/*/lib/{}", llvm_config, subpath);
```

Comment:
Instead of shelling out to `ls` and such, can this just use `read_dir` and syscalls manually here?